rust/parquet/src/record/api.rs

```diff
@@ -102,6 +107,16 @@ macro_rules! row_complex_accessor {
   }
 }
 
+impl RowFormatter for Row {
+    /// Get Display reference for a given field.
+    fn fmt(&self, i: usize) -> Result<&fmt::Display> {
```

Comment:
Instead of returning `Result`, can we just return `&fmt::Display`? I think it is OK to panic here as people should know the number of fields from the `Row` (via `len()`) and pass proper `i` into the method. We can avoid many `unwrap`s in this way.